spec/controllers/ansible_repository_controller_spec.rb

```diff
@@ -97,19 +121,82 @@
       end
     end
 
-    context 'refreshing an actual repository' do
-      let(:params) { {:pressed => "ansible_repository_reload"} }
-      let(:display) { "main" }
+    context 'refreshing an actual repository page' do
+      let(:params) { {:action => "button", :pressed => "ansible_repository_reload"} }
 
       before do
-        controller.instance_variable_set(:@display, display)
+        controller.instance_variable_set(:@breadcrumbs, [])
       end
 
-      it 'calls show and render methods' do
-        expect(controller).to receive(:show)
-        expect(controller).to receive(:render)
+      it 'uses @display over params[:display]' do
+        controller.instance_variable_set(:@display, 'main')
+        receives_init_show
+        expect(controller).to receive(:render).with(:update)
+        controller.send(:button)
+        expect(controller.instance_variable_get(:@_params)[:display]).to eq('main')
+      end
+
+      it 'calls show_output method' do
+        controller.params[:display] = 'output'
```

Comment:
I see that you set `controller.params[:display]` in more `it` blocks exactly like this. You can use `context` blocks and set `params[:display]` in them by using `let` for lazy evaluation and `it` blocks then. For example like this:
```
context 'calling show_output method' do
  let(:display) { 'output' }

   it 'calls show_output method' do
     receives_init_show
     expect(controller).to receive(:show_output)
     renders
     controller.send(:button)
  end
end

``` 
If you do so, you may also want to change line 125 https://github.com/ManageIQ/manageiq-ui-classic/pull/3762/files#diff-62ca34af6656484e37abce1fb2fe1461R125 like this, to set `params[:display]` for the spec properly:
```
      let(:params) { {:action => "button", :pressed => "ansible_repository_reload", :display => display} }
```